packages/@n8n/config/src/decorators.ts

```diff
@@ -84,18 +94,21 @@ export const Nested: PropertyDecorator = (target: object, key: PropertyKey) => {
 };
 
 export const Env =
-	(envName: string): PropertyDecorator =>
+	(envName: string, schema?: PropertyMetadata['schema']): PropertyDecorator =>
 	(target: object, key: PropertyKey) => {
 		const ConfigClass = target.constructor;
 		const classMetadata =
 			globalMetadata.get(ConfigClass) ?? new Map<PropertyKey, PropertyMetadata>();
+
 		const type = Reflect.getMetadata('design:type', target, key) as PropertyType;
-		if (type === Object) {
+		const isEnum = schema && schema instanceof z.ZodEnum;
```

Comment:
`instanceof` works on `undefined` values.

```suggestion
		const isEnum = schema instanceof z.ZodEnum;
```